### PR TITLE
[AC-7811] P1 Fix Internal server error when canceling a non existing office hour reservation as finalist

### PR DESCRIPTION
**Pull request by geofrocker**
### Changes introduced in: [AC-7811](https://masschallenge.atlassian.net/browse/AC-7811) :
- Fix potential check_object_permissions failure with office-hour set to None
### How to test on test2
- masquerade as any finalist
- Make a post request on the CancelOfficeHourReservationView review with a non existing officehour id.
- Get the message `The specified office hour was not found.`

### CODE REVIEW

**Comment by iankigen** [comment]
@geofrocker we have a failing test

File: web/impact/impact/tests/test_cancel_office_hour_reservation.py [hunk 1]
```diff
@@ -119,7 +116,19 @@ def test_user_attempts_to_cancel_non_existent_reservation(self):
         office_hour = MentorProgramOfficeHourFactory(finalist=None)
         response = self._submit_cancellation(office_hour,
                                              user=self.make_user())
-        notification = {'detail': DEFAULT_PERMISSION_DENIED_DETAIL}
+        notification = {
+            'detail': NO_SUCH_RESERVATION,
+            'header': FAIL_HEADER,
+            'success': False}
+        self.assertEqual(response.data, notification)
```
Comment (iankigen) [inline web/impact/impact/tests/test_cancel_office_hour_reservation.py]: can we use the existing `self.assert_ui_notification(response, expected_success, expected detail)` format?

**Review by iankigen — APPROVED** [review]
Get the message `The specified office hour was not found` when trying to cancel a non-existing office hour as a finalist. ✅

**Review by pbx — CHANGES_REQUESTED** [review]
To save us merge conflicts when both this and AC-7469 (the New Monolith long-running branch) are merged, please update the imports in `test_cancel_office_hour_reservation.py` to match the relative form used there:
* https://github.com/masschallenge/impact-api/blob/AC-7469/web/impact/impact/tests/test_cancel_office_hour_reservation.py

**Review by pbx — APPROVED** [review]

**Review by jonkiparsky — APPROVED** [review]

**Pull request merged**